### PR TITLE
Lucas

### PR DESCRIPTION
**Pull request by Geotricity**
Constants class:
- Added Controller Port number
- Added constants to control robot movement

Driver Class:
- Imported commands
- Added Dualshock controller
-  Added methods to get throttle and steer

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java
Comment (danielzyy) [inline src/main/java/frc/robot/Constants.java]: Is this deadzone applied directly to the raw controller values and not the speed values(like from `getY()` and `getX()`)?  If not, and it is meant to be applied on the actual motor speed, I would be wondering why the deadzone is half of the `MAX_THROTTLE_SPEED`, since that seems pretty large of a deadzone.

File: src/main/java/frc/robot/controls/Driver.java
Comment (bricktrooper) [inline src/main/java/frc/robot/controls/Driver.java]: If you want, you could do these all in one line:
```
double throttleValue = -dualshock.getY() * Constants.MAX_THROTTLE_SPEED;
```
I think it might run a little faster unless the Java compiler is super smart :P

File: src/main/java/frc/robot/controls/Driver.java
Comment (bricktrooper) [inline src/main/java/frc/robot/controls/Driver.java]: Same comment as above

File: src/main/java/frc/robot/controls/Driver.java
Comment (danielzyy) [inline src/main/java/frc/robot/controls/Driver.java]: Be careful here, you've already updated `throttleValue` to be a fraction of `MAX_THROTTLE_SPEED` which I am assuming is the speed you are setting to the motors, but you are comparing it to the controller deadzone? If that `DEADZONE` is referring to the raw controller value from `dualshock.getY()`, you should be checking that in a separate if-statement before line 61 to immediately return if controller reading is in deadzone Ex `if(Math.abs(dualshock.getY()) < Constants.DEADZONE) return 0;`

Otherwise, right now the robot will not move until you hit a throttleValue of more than 0.15

File: src/main/java/frc/robot/controls/Driver.java
Comment (bricktrooper) [inline src/main/java/frc/robot/controls/Driver.java:24]: Is dualshock a joystick or an Xbox/playstation controller?  You may want to ask @droid-101 what kind of controller he will be driving with.  If dualshock is used by the operator then you may want to move this code to `Operator.java`.

File: src/main/java/frc/robot/controls/Driver.java
Comment (danielzyy) [inline src/main/java/frc/robot/controls/Driver.java]: Same thing here check previous comment for line 64, are you checking _controller deadzone_ or actual _motor deadzone_?

File: src/main/java/frc/robot/Constants.java
Comment (bricktrooper) [inline src/main/java/frc/robot/Constants.java]: You can actually just put these constants inside `Driver.java` and `Operator.java` right at the top of the classes.  This year we decided to get rid of `Constants.java` and sort the constants into which ever file they are used in.  Keeps it more organized this way :)

File: src/main/java/frc/robot/controls/Driver.java
Comment (danielzyy) [inline src/main/java/frc/robot/controls/Driver.java]: set the `throttleValue` directly to `-dualshock.getY() * Constants.MAX_THROTTLE_SPEED` in one line instead of 2 for clarity.

File: src/main/java/frc/robot/controls/Driver.java
Comment (danielzyy) [inline src/main/java/frc/robot/controls/Driver.java]: set the `steerValue` directly to `dualshock.getZ() * Constants.MAX_STEER_SPEED` in one line instead of 2 for clarity.

**Review by danielzyy — CHANGES_REQUESTED** [review]
Review for controller deadzone and steerValue and throttleValue deadzone condition.

**Review by bricktrooper — CHANGES_REQUESTED** [review]
This looks pretty good so far, just some small things to consider.  Also it seems like your code is failing to compile.  Try building it on your VSCode and see if you can find out what's wrong.  You can also check the "details" next to the build checks and see the compile log.

File: src/main/java/frc/robot/controls/Driver.java
Comment (bricktrooper) [reply src/main/java/frc/robot/controls/Driver.java]: This is a good point!  Maybe we should be checking the deadzone before applying the scaling factors.

**Comment by bricktrooper** [comment]
Also just a tip for the future, if you are also working on the subsystems such as `Intake.java`, best to keep it in a separate PR just to keep things organized.

File: src/main/java/frc/robot/controls/Driver.java
Comment (Geotricity) [reply src/main/java/frc/robot/controls/Driver.java:24]: The dualshock is a generic playstation controller. I'm not yet sure what kind of controller @droid-101 wants to use, but I can always move the code over in a later pull request.

File: src/main/java/frc/robot/controls/Driver.java
Comment (Geotricity) [reply src/main/java/frc/robot/controls/Driver.java]: I'm not able to do this all on one line since the deadzone check has been moved to between reading the value from the joystick and scaling the input.

File: src/main/java/frc/robot/Constants.java
Comment (Geotricity) [reply src/main/java/frc/robot/Constants.java]: You're right, it is a pretty large deadzone. I've changed the code so that it is applied on the raw controller values.

File: src/main/java/frc/robot/controls/Driver.java
Comment (bricktrooper) [reply src/main/java/frc/robot/controls/Driver.java:24]: Sounds good to me!

**Comment by Geotricity** [comment]
I've made the changes requested here in the newest commit by:

- Removing Constants.java and placed constants in the classes where they are used
- Changing the DEADZONE check to occur before the joystick input is scaled
- Commenting out a line in ExampleCommand that seemed to be causing the compilation error

**Comment by Geotricity** [comment]
> Also just a tip for the future, if you are also working on the subsystems such as `Intake.java`, best to keep it in a separate PR just to keep things organized.

Thanks, I'll do that next time. :)

File: src/main/java/frc/robot/controls/Driver.java
Comment (Geotricity) [reply src/main/java/frc/robot/controls/Driver.java:24]: Ok, I heard from @droid-101 that he wants 2 joysticks for the driver and a controller for the operator. Should I change that here or do that in a different pull request?

**Review by bricktrooper — APPROVED** [review]
Looks good man, I think you can merge this for now unless anyone else has any comments.  Also make sure to change update your PR title when merging and just replace the auto-generated description with the nice bullet points you wrote at the top.  btw you might have to hit "resolve" on all the comments before it lets you merge.

**Review by danielzyy — APPROVED** [review]
Looks good to me

**Pull request merged**
